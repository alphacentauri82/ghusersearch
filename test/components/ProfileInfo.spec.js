import React from 'react';
import renderer from 'react-test-renderer';
import {
  StyleSheetTestUtils,
} from 'aphrodite';
import ProfileInfo from 'components/ProfileInfo';

describe('Component: UserInfo', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render only the props that are present', () => {
    const component = renderer.create(
      <ProfileInfo
        company="@cool place"
        location="London"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render nothing if all props are null', () => {
    const component = renderer.create(
      <ProfileInfo
        blog={null}
        location={null}
        company={null}
      />
    );
    expect(component).toMatchSnapshot();
  });

});

