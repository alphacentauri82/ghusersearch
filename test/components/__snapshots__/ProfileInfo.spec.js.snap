// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: UserInfo should render nothing if all props are null 1`] = `null`;

exports[`Component: UserInfo should render only the props that are present 1`] = `
<ul>
  <li
    className="UserInfo_item_1bldfdq u-flex u-flexAlignItemsCenter"
  >
    <svg />
    <p
      className="UserInfo_text_f9e8qy"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<a href=\\"http://github.com/cool\\">@cool</a> place",
        }
      }
    />
  </li>
  <li
    className="UserInfo_item_1bldfdq u-flex u-flexAlignItemsCenter"
  >
    <svg />
    <p
      className="UserInfo_text_f9e8qy"
    >
      London
    </p>
  </li>
</ul>
`;
